articles/implementing-mvvm-architecture-in-android-using-kotlin/index.md

```diff
@@ -0,0 +1,71 @@
+
+### Introduction
+Let's start by evaluating what android architectures existed before MVVM. The first component is Model View Presenter denoted by MVP. Though this architecture separates the business logic from the app's UI, it is not easy to implement. In the long-run, this can translate into high development costs. The second android architecture is [MVC](https://openclassrooms.com/en/courses/4661936-develop-your-first-android-application/4679186-learn-the-model-view-controller-pattern#:~:text=Most%20Android%20developers%20use%20a,apply%20to%20our%20TopQuiz%20application.). Just like MVP, it is also quite complex and not suitable for small projects. Google introduced MVVM (Model-View-ViewModel)  to resolve these challenges. By separating code into smaller chunks, MVVM simplifies the debugging process. Through this article, you'll understand MVVM architecture and implement it in an application. Furthermore, this article shows how to debug common errors resulting from this architecture. You can learn more about MVVM [here](https://developer.android.com/topic/libraries/architecture/viewmodel?gclid=Cj0KCQiA7qP9BRCLARIsABDaZzhDtIsNoyAcuVYiA3F3smhaKd4THplNIp1nDr-KGB_XWkzZxiIvrVAaAjYKEALw_wcB&gclsrc=aw.ds).
+
+Let's dive in!
+
+
+### Prerequisites
+1. Android studio
+2. You must be familiar with [Kotlin](https://developer.android.com/kotlin/campaign/learn?gclid=Cj0KCQiA7qP9BRCLARIsABDaZzh1wodOJn7w8kKTtWq8yNFlx9xoqzEE_cU2KkCO2Ecdyyr2frGOVjQaAlSuEALw_wcB&gclsrc=aw.ds)
+3. Install [lifecycle](https://developer.android.com/jetpack/androidx/releases/lifecycle) dependencies
+4. Download the start code from [here](https://github.com/WanjaMIKE/MVVMExample)
+
+#### The goal of the tutorial
+By the end of this tutorial, you will create an app that takes input and displays it on a recycler view. A screenshot of the app is shown below.
+
+### Step 1 – Launching Android Studio.
+Launch Android Studio and create a new project, as shown below. Make sure that you select Kotlin as your preferred programming language. If you do not have Android Studio, you can install it from [here](https://developer.android.com/studio?gclid=Cj0KCQiA7qP9BRCLARIsABDaZzieBJWjBnokDdH6b0gQchoqudRXNohAGp_noSqALLuSlYuwA6EB5T4aAntwEALw_wcB&gclsrc=aw.ds)
+
+![getting started](/engineering-education/implementing-mvvm-architecture-in-android-using-kotlin/getting-started.png)
+
+### Step 2 – Creating the model.
+Create the app model. This is also referred to as the data class. To avoid confusion, create a package named model inside the java folder. Then, create a data class named Blog in the model package as shown below. For simplicity, the data class will only have one variable (title). In Kotlin, there is no need to use the get and set keywords. These functionalities are added to the data class automatically.
```

Comment:
Rephrase the last sentence to this "There is no need to add getters and setters, Kotlin adds them to the class automatically." It is shorter and also eliminates the use of passive voice.